### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by mcrowson**
Adds Rooibos testing framework and changes deploy/testing away from shell scripts to makefile.

**Changes**

Previously used some top level shell scripts that have been absorbed into the makefile.

`make install` deploys the roku app
`make test` runs the tests on the roku

**Issues**

#35

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,17 +20,22 @@ sh make_images.sh
...
-And then
-```
-export ROKU_DEV_TARGET=<your roku IP>
-export ROKU_DEV_PASSWORD=<your roku password>
-```
\ No newline at end of file
+### Testing
+Testing is done with the [Rooibos](https://github.com/georgejecook/rooibos/) library. This works by including the tests in the deployment and then looking at telnet for the test results.
+
+`make tests` will package up the application and tests and the deploy it to the Roku. Test results can be seen via `telnet ${ROKU_DEV_TARGET} 8085`
```
Comment (bisby) [inline README.md]: this should be `make test` singular (or the makefile should be changed)

File: README.md [hunk 2]
```diff
@@ -20,17 +20,22 @@ sh make_images.sh
...
+This is the IP address of your roku and the password you set for the rokudev account when you put your device in developer mode.
 
-And then
-```
-export ROKU_DEV_TARGET=<your roku IP>
-export ROKU_DEV_PASSWORD=<your roku password>
-```
\ No newline at end of file
+### Testing
+Testing is done with the [Rooibos](https://github.com/georgejecook/rooibos/) library. This works by including the tests in the deployment and then looking at telnet for the test results.
```
Comment (bisby) [inline README.md]: `make test` has a dependency on rooibosPreprocessor

Which I appear to be having some issues with at the the moment.

Comment (mcrowson) [reply README.md]: Is it this issue? https://github.com/georgejecook/rooibosPreprocessor/issues/5

Comment (bisby) [reply README.md]: That's the one. Just means I haven't been able to test yet. 👍

Comment (mcrowson) [reply README.md]: Take a look at bottom comment, just need to rename a file as the case syntax got messed up somewhere.

**Comment by bisby** [comment]
This all looks good.  My last question is that I've had a few people struggle to get this up and running even with provided bash scripts. Is `make` universally available enough to allow people to begin testing with it.

Or would the answer become "if you don't feel comfortable with make, you should hold off until this is in the Roku Store"

**Comment by mcrowson** [comment]
I would guess that people wanting to learn/help wouldn't have trouble learning how to install `make`. If they do we can help, but the idea is that make takes care of the build details making it even easier. So maybe make issues are simply documentation issues for us going forward to help people get setup.

**Review by bisby — APPROVED** [review]

**Pull request merged**